src/main/java/com/fasterxml/jackson/databind/deser/BeanDeserializerFactory.java

```diff
@@ -977,8 +977,12 @@ protected SettableBeanProperty constructSettableProperty(DeserializationContext
                     beanDesc.getClassAnnotations(), (AnnotatedMethod) mutator);
         } else {
             // 08-Sep-2016, tatu: wonder if we should verify it is `AnnotatedField` to be safe?
-            prop = new FieldProperty(propDef, type, typeDeser,
-                    beanDesc.getClassAnnotations(), (AnnotatedField) mutator);
+            AnnotatedField field = (AnnotatedField) mutator;
+            if (field.isImmutable()) {
```

Comment:
Comment sounds like all immutable fields would be dropped, but it really is only about records.
So instead of calling `Field` to ask, and `Field` having to find its declaring class, to check if that declaring class is a record (having to do type refinement), should check here: something like

```
            if (beanDesc.getType().isRecordType()) {
                return null;
            }
```

Reply:
Made the change.